src/pages/dataset-list-view/dataset-list-view.js

```diff
@@ -1,170 +1,245 @@
-import LoginZone from '../../components/login-zone'
-import FacetSidebar from '../../components/facet-sidebar'
-import ErrorComponent from '../../components/generic-elements/error-component'
-import LoadingElement from '../../components/generic-elements/loading-element'
-import Checkbox from '../../components/generic-elements/checkbox'
-import qs from 'qs'
-import _ from 'lodash'
-import { QueryStringBuilder } from '../../utils'
+import "./dataset-list-view.scss";
+import { Component } from "react";
+import DatasetList from "../../components/dataset-list";
+import Paginator from "../../components/generic-elements/paginator";
+import Select from "../../components/generic-elements/select";
+import Search from "../../components/generic-elements/search";
+import LoginZone from "../../components/login-zone";
+import FacetSidebar from "../../components/facet-sidebar";
+import ErrorComponent from "../../components/generic-elements/error-component";
+import LoadingElement from "../../components/generic-elements/loading-element";
+import Checkbox from "../../components/generic-elements/checkbox";
+import qs from "qs";
+import _ from "lodash";
 
 export default class extends Component {
   constructor(props) {
-    super(props)
-    this.state = { currentPage: 1, pageSize: 10 }
+    super(props);
   }
 
   componentDidMount() {
-    this.props.fetchData(this.pageNumber, this.state.pageSize, this.sort, this.searchParams, this.facets, this.apiAccessible)
-  }
-
-  componentDidUpdate() {
-    window.onpopstate = (e) => {
-      this.props.fetchData(this.pageNumber, this.state.pageSize, this.sort, this.searchParams, this.facets, this.apiAccessible)
+    this.updateDatasetSearchParamsFromQuery();
+    this.fetchData(this.props.searchParams);
+  }
+
+  componentDidUpdate(previousProps) {
+    const urlSearchString = this.props.location.search;
+    const propSearchString =
+      "?" +
+      this.queryString({
+        ...this.props.searchParams,
+        page: this.props.pageNumber
+      });
+
+    const previousPropSearchString =
+      "?" +
+      this.queryString({
+        ...previousProps.searchParams,
+        page: previousProps.pageNumber
+      });
+
+    const stateAndUrlOutOfSync = propSearchString !== urlSearchString;
+    const stateWasUpdated = propSearchString !== previousPropSearchString;
+
+    if (stateAndUrlOutOfSync) {
+      if (stateWasUpdated) {
+        //update url bar to match props
+        this.updateQueryParameters({
+          ...this.props.searchParams,
+          page: this.props.pageNumber
+        });
+      } else {
+        //update redux state to match url bar
+        this.updateDatasetSearchParamsFromQuery();
+        this.fetchData(this.props.searchParams);
+      }
     }
   }
 
-  onPageChange(page) {
-    this.setState({ currentPage: page })
-    this.refreshDatasets(this.searchParams, this.sort, this.facets, page, this.apiAccessible)
+  updateDatasetSearchParamsFromQuery() {
```

Comment:
It would be nice to have the `updateQueryParameters` function next to this one for clarity, but it's NBD overall